codespeed/views.py

```diff
@@ -731,3 +739,239 @@ def add_json_results(request):
+        baseline_proj = project
+        baseline_exe = executable
+        baseline_branch = branch
+
+        try:
+            if 'base_env' in data:
+                baseline_env = Environment.objects.get(name=data['base_env'])
+            if 'base_proj' in data:
+                baseline_proj = Project.objects.get(name=data['base_proj'])
+            if 'base_exe' in data:
+                baseline_exe = Executable.objects.get(name=data['base_exe'],
+                                                      project=baseline_proj)
+            if 'base_branch' in data:
+                baseline_branch = Branch.objects.get(name=data['base_branch'],
+                                                     project=baseline_proj)
+        except Environment.DoesNotExist:
+            return None, HttpResponseNotFound(
+                            'Baseline environment "' + data['base_env'] +
+                            '" does not exist!')
+        except Project.DoesNotExist:
+            return None, HttpResponseNotFound(
+                            'Baseline project "' + data['base_proj'] +
+                            '" does not exist!')
+        except Executable.DoesNotExist:
+            return None, HttpResponseNotFound(
+                            'Baseline executable "' + data['base_exe'] +
+                            '"does not exist!')
+        except Branch.DoesNotExist:
+            return None, HttpResponseNotFound(
+                            'Baseline branch "' + data['base_branch'] +
+                            '" does not exist!')
+
+        base_data = Result.objects.get(
+                                benchmark=benchmark,
+                                environment=baseline_env,
+                                executable=baseline_exe,
+                                revision__project=baseline_proj,
+                                revision__branch=baseline_branch,
+                                revision__commitid=baseline_commit_name)
+
+        if base_data:
+            ref_value = base_data.value
+        else:
+            return None, HttpResponseNotFound(
+                            'Result for revision "' + baseline_commit +
+                            '" does not exist !')
+
+    if relative_results:
+        for element in result_list:
+            element.value = (100 * (element.value - ref_value)) / ref_value
+
+    return {
+            'environment': environment,
+            'project': project,
+            'executable': executable,
+            'branch': branch,
+            'benchmark': benchmark,
+            'results': result_list,
+            'relative': relative_results,
+           }, None
+
+
+@require_GET
+def makeimage(request):
+    data = request.GET
+
+    err_msg, err = validate_results_request(data)
+
+    if err:
+        return HttpResponseBadRequest(err_msg)
+
+    result_data, error = get_benchmark_results(data)
+
+    if error is not None:
+        return error
```

Comment:
With the changes suggested this would be:

```python
try:
    validate_results_request(data)
except ValidationError as e:
    return HttpResponseBadRequest(str(e))

try:
     result_data = get_benchmark_results(data)
except ObjectDoesNotExist as e:
    return HttpResponseNotFound(str(e))
```